packages/common/src/permissions/_internal/checkAssertions.ts

```diff
@@ -18,9 +18,29 @@ export function checkAssertions(
   // Only return a check if the policy is defined
   if (policy.assertions) {
     // iterate over the assertions, creating a check for each entry
-    checks = policy.assertions.map((assertion: IPolicyAssertion) => {
-      return checkAssertion(assertion, entity, context);
-    });
+    checks = policy.assertions.reduce(
+      (acc: IPolicyCheck[], assertion: IPolicyAssertion) => {
+        let conditionResult = true;
```

Comment:
minor, but I might suggest naming this something like `shouldCheckAssertion` to make it a bit more clear?